op-program/client/l2/engineapi/block_processor.go

```diff
@@ -138,12 +139,7 @@ func (b *BlockProcessor) CheckTxWithinGasLimit(tx *types.Transaction) error {
 func (b *BlockProcessor) AddTx(tx *types.Transaction) error {
 	txIndex := len(b.transactions)
 	b.state.SetTxContext(tx.Hash(), txIndex)
-
-	context := core.NewEVMBlockContext(b.header, b.dataProvider, nil, b.dataProvider.Config(), b.state)
-	vmConfig := *b.dataProvider.GetVMConfig()
-	// TODO(#14038): reuse evm
-	evm := vm.NewEVM(context, b.state, b.dataProvider.Config(), vmConfig)
-	receipt, err := core.ApplyTransaction(evm, b.gasPool, b.state, b.header, tx, &b.header.GasUsed)
+	receipt, err := core.ApplyTransaction(b.evm, b.gasPool, b.state, b.header, tx, &b.header.GasUsed)
```

Comment:
LGTM.

`ApplyTransaction` -> `ApplyTransactionExtended` -> `ApplyTransactionWithEVM` -> `ApplyMessage` -> `evm.SetTxContext(NewEVMTxContext(msg))` (which replaced the original `evm.Reset` func in latest geth release)

So it should be safe to reuse the EVM instance. And vmConfig stays the same across the block (including precompile overrides etc.)